src/patients/edit/EditPatient.tsx

```diff
@@ -81,27 +85,55 @@ const EditPatient = () => {
     if (!patient.givenName) {
       inputIsValid = false
       setErrorMessage(t('patient.errors.updatePatientError'))
-      setInvalidFields((prevState) => ({
-        ...prevState,
+      setInvalidFields({
+        ...invalidFields,
         givenName: true,
-      }))
-      setFeedbackFields((prevState) => ({
-        ...prevState,
+      })
+      setFeedbackFields({
+        ...feedbackFields,
         givenName: t('patient.errors.patientGivenNameFeedback'),
-      }))
+      })
     }
     if (patient.dateOfBirth) {
       if (parseISO(patient.dateOfBirth) > new Date(Date.now())) {
         inputIsValid = false
         setErrorMessage(t('patient.errors.updatePatientError'))
-        setInvalidFields((prevState) => ({
-          ...prevState,
+        setInvalidFields({
+          ...invalidFields,
           dateOfBirth: true,
-        }))
-        setFeedbackFields((prevState) => ({
-          ...prevState,
+        })
+        setFeedbackFields({
+          ...feedbackFields,
           dateOfBirth: t('patient.errors.patientDateOfBirthFeedback'),
-        }))
+        })
+      }
+    }
+    if (patient.phoneNumber) {
+      if (!patient.phoneNumber.match(/^\d+/)) {
+        inputIsValid = false
+        setErrorMessage(t('patient.errors.updatePatientError'))
+        setInvalidFields({
+          ...invalidFields,
+          phoneNumber: true,
+        })
+        setFeedbackFields({
+          ...feedbackFields,
+          phoneNumber: t('patient.errors.patientPhoneNumberFeedback'),
+        })
+      }
+    }
+    if (patient.email) {
+      if (!patient.email.match(/^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{2,3})+$/)) {
+        inputIsValid = false
+        setErrorMessage(t('patient.errors.updatePatientError'))
+        setInvalidFields({
+          ...invalidFields,
+          email: true,
+        })
+        setFeedbackFields({
+          ...feedbackFields,
+          email: t('patient.errors.patientEmailFeedback'),
+        })
```

Comment:
I think we may want to use https://www.npmjs.com/package/validator (or equivalent library if there is a different one). 

I fear that there are edge cases that we have not considered. For example, phone numbers can have (, ), -, / in some cases.

This library seems comprehensive in the validations it performs.

Reply:
@fox1t thoughts here?

Reply:
Yes, we absolutely need to relay on some external library to do this properly. However I don't have any preference.